### DISCUSSION TITLE
Reducing bundle size

### DISCUSSION THREAD

**Discussion by perryd01** [UI5/webcomponents#10102] [Q&A] [answered]
are there any tricks to reduce the bundle size of a react application that is using ui5?

Looking at the logs of the [Vite starter](https://stackblitz.com/github/SAP/ui5-webcomponents-react/tree/main/templates/vite-ts?file=src%2FApp.tsx) build it shows that the the "main" chuck is already over the recommended size limit and there are way to many i18n related files bundled, even if they are lazy loaded and will not be imported on the client.

```txt
dist/assets/lt-BVWqkFr_.js                             115.99 kB │ gzip:  25.47 kB
dist/assets/cs-DcM44PeB.js                             116.21 kB │ gzip:  25.22 kB
dist/assets/ar-CUjm9SSI.js                             137.19 kB │ gzip:  29.41 kB
dist/assets/ar_EG-DQlFzN6x.js                          137.19 kB │ gzip:  29.41 kB
dist/assets/ar_SA-lp21-ovF.js                          137.26 kB │ gzip:  29.53 kB
dist/assets/index-CWCp_t8N.js                          768.58 kB │ gzip: 178.11 kB

(!) Some chunks are larger than 500 kB after minification. Consider:
- Using dynamic import() to code-split the application
- Use build.rollupOptions.output.manualChunks to improve chunking: https://rollupjs.org/configuration-options/#output-manualchunks
- Adjust chunk size limit for this warning via build.chunkSizeWarningLimit.
✓ built in 19.04s
```

**Comment by Lukas742** [comment, marked as answer]
Hi @perryd01 we looked into this issue and can confirm that the message-bundle files (translations) are bundled in Base64 encoded, which can affect the bundle/chunk size. Since assets are mostly implemented by the ui5 web components team, I'll forward this question to their repo.
